[go] Update B3 header environment variable documentation (#16351)
DataDog/documentation: content/en/tracing/trace_collection/custom_instrumentation/go.md

supported: `Datadog` and `B3`.

Configure injection styles using the environment variable:
`DD_TRACE_PROPAGATION_STYLE_INJECT=Datadog,B3`

Configure extraction styles using the environment variable:
`DD_TRACE_PROPAGATION_STYLE_EXTRACT=Datadog,B3`

Configure both styles using the environment variable:
`DD_TRACE_PROPAGATION_STYLE=Datadog,B3`.

> Note that if either `DD_TRACE_PROPAGATION_STYLE_INJECT` or 
> `DD_TRACE_PROPAGATION_STYLE_EXTRACT` are set, those will take precedence.

Reviewer: ```suggestion
*Note*: `DD_TRACE_PROPAGATION_STYLE_INJECT` and `DD_TRACE_PROPAGATION_STYLE_EXTRACT`, if set, take precedence.
```

supported: `Datadog` and `B3`.

Configure injection styles using the environment variable:
`DD_TRACE_PROPAGATION_STYLE_INJECT=Datadog,B3`

Configure extraction styles using the environment variable:
`DD_TRACE_PROPAGATION_STYLE_EXTRACT=Datadog,B3`

> Configure both styles using the environment variable:
> `DD_TRACE_PROPAGATION_STYLE=Datadog,B3`
> 
> Values from the respective injection and extraction style environment variables
> take precedence over `DD_TRACE_PROPAGATION_STYLE`.

Reviewer 2: ```suggestion
Configure both styles using the environment variable:
`DD_TRACE_PROPAGATION_STYLE=Datadog,B3`.
Note that if either `DD_TRACE_PROPAGATION_STYLE_INJECT` or 
`DD_TRACE_PROPAGATION_STYLE_EXTRACT` are set, those will take precedence.
```

Reviewer 2: Just a slight wording suggestion.

Author: Thanks!
